File: 2150/2150.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
#define FOR(i, a, n) for(int i = a; i < n; i++)
#define REP(i, n) FOR(i, 0, n)
#define PB push_back

using namespace std;

vector<bool> prime;

void sieve(int n) {
  prime.assign(n + 1, true);
  prime[0] = false, prime[1] = false;
  int s = sqrt(n) + 1;
  REP(i, s) {
    if(!prime[i]) continue;
    for(int j = i * 2; j < n + 1; j += i)
      prime[j] = false;
  }
}

int solve(int n, int p) {
  vector<int> a, v;
  for(int i = n + 1; a.size() < p; i++)
    if(prime[i]) a.PB(i);
  REP(i, a.size()) FOR(j, i, a.size())
    v.PB(a[i] + a[j]);
  sort(v.begin(), v.end());
  return v[p - 1];
}

int main() {
  int n, p;
  sieve(200000);
  while(cin >> n >> p, n != -1 || p != -1)
    cout << solve(n, p) << endl;
  return 0;
}

File: 2150/2150.py
# -*- coding: utf-8 -*-

import math

prime = []

def sieve(n):
    global prime
    prime = [True] * (n + 1)
    prime[0] = False
    prime[1] = False
    s = int(math.sqrt(n)) + 1
    for i in range(s):
        if not prime[i]:
            continue
        for j in range(i * 2, n + 1, i):
            prime[j] = False

def solve(n, p):
    global prime
    a = []
    cnt = n + 1
    while len(a) < p:
        if prime[cnt]:
            a.append(cnt)
        cnt += 1
    v = []
    for i in range(len(a)):
        for j in range(i, len(a)):
            v.append(a[i] + a[j])
    v.sort()
    return v[p - 1]

if __name__ == "__main__":
    sieve(200000)
    while True:
        (n, p) = map(int, input().split())
        if n == -1 and p == -1:
            break
        print(solve(n, p))
